packages/libs/error-stack/tests/common.rs

```diff
@@ -175,3 +179,132 @@ pub fn messages<E>(report: &Report<E>) -> Vec<String> {
 pub fn frame_kinds<E>(report: &Report<E>) -> Vec<FrameKind> {
     report.frames().map(Frame::kind).collect()
 }
+
+#[cfg(all(nightly, feature = "std"))]
+pub fn supports_backtrace() -> bool {
+    // we need to track 3 states:
+    //  1) -1 (not checked)
+    //  2) 0 (disabled)
+    //  3) 1 (enabled)
+    static STATE: AtomicI8 = AtomicI8::new(-1);
+
+    match STATE.load(Ordering::SeqCst) {
+        -1 => {
+            let bt = std::backtrace::Backtrace::capture();
+            if bt.status() == std::backtrace::BacktraceStatus::Captured {
+                STATE.store(1, Ordering::SeqCst);
+                true
+            } else {
+                STATE.store(0, Ordering::SeqCst);
+                false
+            }
+        }
+        0 => false,
+        1 => true,
+        _ => unreachable!(),
+    }
```

Comment:
Not blocking the PR but I think using a once-cell is easier here.

Reply:
Yes, but OnceCell (and Lazy) are either unstable features or require the hooks feature. I could add `once-cell` as a dev-dependency tho. (didn't want to introduce a new dependency)

Reply:
As we already depend on `once-cell` it's fair to use it here as well. TBH `once-cell` is pretty simple in comparison to other dev-dependencies we use (esp. tracing is very heavy).